### PR TITLE
PMM-7737 Generate password for each user.

### PR DESCRIPTION
**Pull request by BupycHuk**
https://github.com/Percona-Lab/pmm-submodules/pull/1793

### CODE REVIEW

File: service/k8sclient/k8sclient.go [hunk 1]
```diff
@@ -560,6 +548,45 @@ func (c *K8sClient) CreateXtraDBCluster(ctx context.Context, params *XtraDBParam
 	return c.kubeCtl.Apply(ctx, res)
 }
 
+func (c *K8sClient) generateXtraDBPasswords() (map[string][]byte, error) {
+	root, err := generatePassword(passwordLength)
```
Comment (oter) [inline service/k8sclient/k8sclient.go:552]: I would suggest to generate 6 passwords using a loop and then just use that array for assigning.

File: service/k8sclient/k8sclient.go [hunk 2]
```diff
@@ -1069,6 +1082,39 @@ func (c *K8sClient) CreatePSMDBCluster(ctx context.Context, params *PSMDBParams)
...
+	}
+	userAdmin, err := generatePassword(passwordLength)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate userAdmin password")
+	}
+
+	// secrets represents stringData part of
+	// https://github.com/percona/percona-xtradb-cluster-operator/blob/main/deploy/secrets.yaml.
+	secrets := map[string][]byte{
+		"MONGODB_BACKUP_USER":              []byte("backup"),
```
Comment (jprukner) [inline service/k8sclient/k8sclient.go:1106]: similar here, I'd set only the password keys in the beginning, doing the same as for generateXtraDBPasswords, with small change -> adding *_USER keys befor returning. We may even reuse some code with couple of more changes. 
Something like
```

func generatePasswords(map[string][]byte) error {...}
```

**Review by jprukner — CHANGES_REQUESTED** [review]
Let's DRY this code :)

**Review by jprukner — APPROVED** [review]

File: service/k8sclient/secrets.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+func generatePSMDBPasswords() (map[string][]byte, error) {
+	// secrets represents stringData part of
+	// https://github.com/percona/percona-xtradb-cluster-operator/blob/main/deploy/secrets.yaml.
+	secrets := map[string][]byte{
+		"MONGODB_BACKUP_USER":          []byte("backup"),
+		"MONGODB_CLUSTER_ADMIN_USER":   []byte("clusterAdmin"),
+		"MONGODB_CLUSTER_MONITOR_USER": []byte("clusterMonitor"),
+		"MONGODB_USER_ADMIN_USER":      []byte("userAdmin"),
+	}
+	passwords, err := generatePasswords(map[string][]byte{
```
Comment (oter) [inline service/k8sclient/secrets.go:80]: it's a bit confusing to me having `[]byte` and not using it. Let's replace it with `map[string]struct{}`.

Comment (BupycHuk) [reply service/k8sclient/secrets.go:80]: I reuse the same map to generate a new one.

**Review by oter — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]

**Pull request merged**